src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/extension.ts

```diff
@@ -124,6 +125,15 @@ export async function activate(vscodeType: typeof vscodeapi, context: ExtensionC
                 documentManager,
                 languageServiceClient,
                 logger);
+            const legend = await languageServiceClient.getSemanticTokenLegend();
+            const semanticTokenProvider = new RazorDocumentSemanticTokensProvider(
+              documentSynchronizer,
+              documentManager,
+              languageServiceClient,
+              logger);
+            if (legend) {
+                localRegistrations.push(vscodeType.languages.registerDocumentSemanticTokensProvider(RazorLanguage.id, semanticTokenProvider, legend));
```

Comment:
I think the LanguageClient that we use for our LangaugeServer does all this. Meaning if we move our language server to use the appropriate endpoints / broadcast the appropriate capabilities I think we don't need to do any client side registering like this:

The language client hookup for semanticness: https://github.com/microsoft/vscode-languageserver-node/blob/4c0963d6fdfd657044092c5d4db61a87b54471ef/testbed/client/src/extension.ts#L51-L52

Extra details on their impl of the languageClient feature: https://github.com/microsoft/vscode-languageserver-node/blob/aa6592ee4b4e0276a2e7c8974e19dcdf7b0a61f9/client/src/semanticTokens.proposed.ts

Reply:
We talked about this offline. Right now I'm kind of thinking it's not worth it to integrate "officially" because none of the libraries we use yet have official support for it, and it's not even in the official LSP spec yet (listed as "Upcoming"), so we'd have to hack together our own capabilities which would eventually be thrown away. I'm going to come at this again when my brain is a bit fresher on Monday and see if I still feel the same way. If we DO decide not to map all the capabilities and what not yet we should prefix our endpoints to make sure they don't collide (so `textDocument/SemanticTokens` would be `_ms_/textDocument/SemanticTokens`), and I should communicate that to our VS counterparts.

Reply:
We're going with the _ms_ option for now due to the spec for this still being in flux.